internal/provider/cloudinitsecret/resource_cloudinitsecret.go

```diff
@@ -100,6 +110,19 @@ func resourceCloudInitSecretDelete(ctx context.Context, d *schema.ResourceData,
 	if err != nil && !apierrors.IsNotFound(err) {
 		return diag.FromErr(err)
 	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{constants.StateImageTerminating, constants.StateCommonActive},
```

Comment:
You're right. This whole block of code doesn't make a lot of sense since the cloudinit secret doesn't have a state. It either exists or it doesn't but there is not any intermediate state as there would be with a Pod or similar. Therefore we don't need to wait here, the `DELETE` request should be enough.